source/simulator/helper_functions.cc

```diff
@@ -2115,7 +2115,8 @@ namespace aspect
                   }
 
                 // ... and change the boundary id of any outflow boundary faces.
-                if (integrated_flow > 0)
+                // If there is no flow, we do not want to apply dirichlet boundary conditions either.
+                if (integrated_flow >= 0)
```

Comment:
Yes, note that this is now the same as in DG:
https://github.com/geodynamics/aspect/blob/cbc7833440cd05c7370bc28f4119199f052dbcfa/source/simulator/assemblers/advection.cc#L780

Reply:
Oh, it's great that it's now consistent between the two methods.